docs/guides/latest/volumesnapshot/deployment.md

```diff
@@ -0,0 +1,525 @@
+apiVersion: snapshot.storage.k8s.io/v1alpha1
+kind: VolumeSnapshotClass
+metadata:
+  annotations:
+    snapshot.storage.kubernetes.io/is-default-class: "true"
+  name: default-snapshot-class
+snapshotter: pd.csi.storage.gke.io
+```
+
+Here,
+
+- `metadata.annotations` annotations are used to set default [volumeSnapshotClass](https://kubernetes.io/blog/2018/10/09/introducing-volume-snapshot-alpha-for-kubernetes/).
+- `snapshotter` field to point to the respective CSI driver that is responsible for taking snapshot. As we are using [GCE Persistent Disk CSI Driver](https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver), we are going to use `pd.csi.storage.gke.io` in this field.
+
+Let's create the `volumeSnapshotClass` crd we have shown above,
+
+```console
+$ kubectl apply -f ./docs/examples/guides/latest/volumesnapshot/default-volumesnapshotclass.yaml
+volumesnapshotclass.snapshot.storage.k8s.io/default-snapshot-class created
+```
+
+To keep everything isolated, we are going to use a separate namespace called `demo` throughout this tutorial.
+
+```console
+$ kubectl create ns demo
+namespace/demo created
+```
+
+>Note: YAML files used in this tutorial are stored in [/docs/examples/guides/latest/volumesnapshot](/docs/examples/guides/latest/volumesnapshot) directory of [stashed/stash](https://github.com/stashed/stash) repository.
+
+## Take Volume Snapshot
+
+Here, we are going to deploy a Deployment with two PVCs and generate some sample data on it. Then, we will take snapshot of those PVCs using Stash.
+
+**Create PersistentVolumeClaim :**
+
+At first, let's create two sample PVCs. We will mount these PVCs in our targeted Deployment.
+
+Below is the YAML of the sample PVCs,
+
+```yaml
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: source-pvc-1
+  namespace: demo
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: standard
+  resources:
+    requests:
+      storage: 6Gi
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: source-pvc-2
+  namespace: demo
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: standard
+  resources:
+    requests:
+      storage: 6Gi
+```
+
+Create the PVCs we have shown above.
+
+```console
+$ kubectl apply -f ./pvcs.yaml
+persistentvolumeclaim/source-pvc-1 created
+persistentvolumeclaim/source-pvc-2 created
+```
+
+**Deploy Deployment :**
+
+Now, we will deploy a Deployment that uses the above PVCs. This Deployment will automatically generate sample data (sample-file.txt file) in `/source/data` directory where we have mounted the desired PVCs.
+
+Below is the YAML of the Deployment that we are going to create,
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: stash-demo
+  name: stash-demo
+  namespace: demo
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: stash-demo
+  template:
+    metadata:
+      labels:
+        app: stash-demo
+      name: busybox
+    spec:
+      containers:
```

Comment:
I think it will be nice to write some data.

`echo sample-file1 > source/data/sample-file1.txt`